### PR TITLE
CSS-2295 Initial structure for the admin charm

### PR DESCRIPTION
**Pull request by frankban**
The admin charm includes the temporal admin tools.
See https://github.com/canonical/temporal-k8s-operator/blob/main/CONTRIBUTING.md#admintemporal for a description about how this is supposed to work together the Temporal charm.

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        """
+        self.unit.status = WaitingStatus(f"handling {event.relation.name} change")
+        database_connections = event.relation.data[event.app].get("database_connections")
+        self._state.database_connections = json.loads(database_connections) if database_connections else None
+        self._setup_db_schemas(event)
+
+    @log_event_handler
+    def _on_tctl_action(self, event):
+        """Run the tctl command line tool."""
+        # TODO(frankban): make this work.
```
Comment (kian99) [inline src/charm.py:78]: Have a look at https://juju.is/docs/sdk/interact-with-pebble#heading--run-commands for running commands with Pebble.

**Review by kian99 — APPROVED** [review]
LGTM

File: src/charm.py (hunk 1 above)
Comment (frankban) [reply src/charm.py:78]: Yes, that's how it should work, thanks!

File: .github/workflows/on_push.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  push:
+    branches:
+      - master
+      - main
+      - track/**
+
+jobs:
+  run-tests:
+    name: Run Tests
+    uses: ./.github/workflows/ci.yaml
```
[the remark this answers is not in the document]
Comment (frankban) [reply .github/workflows/on_push.yaml:12]: I'll do on next branch.

File: charmcraft.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+# See LICENSE file for licensing details.
+
+# Learn more about charmcraft.yaml configuration at:
+# https://juju.is/docs/sdk/charmcraft-config
+
+type: "charm"
+bases:
+  - build-on:
+    - name: "ubuntu"
+      channel: "20.04"
```
Comment (ijansky) [inline charmcraft.yaml:11]: Could we use 22.04?

File: metadata.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+# For a complete list of supported options, see:
+# https://juju.is/docs/sdk/metadata-reference
+
+name: temporal-admin-k8s
```
Comment (ijansky) [inline metadata.yaml:7]: Should there be the "-operator" suffix?

**Review by ijansky — APPROVED** [review]

**Review by alesstimec — APPROVED** [review]

File: charmcraft.yaml (hunk 3 above)
Comment (frankban) [reply charmcraft.yaml:11]: Yes, I'll update them all as a task later.

File: metadata.yaml (hunk 4 above)
Comment (frankban) [reply metadata.yaml:7]: No, according to naming conventions: https://juju.is/docs/sdk/naming#heading--naming-charms
I guess sthe "-operator" suffix is used mostly for repo names, but not charm names.

**Pull request merged**